nix/docker.nix

```diff
@@ -0,0 +1,83 @@
+############################################################################
+# Docker image builder
+#
+# To test it out, use:
+#
+#   docker load -i $(nix-build -A dockerImage --no-out-link)
+#   docker run cardano-node
+#
+############################################################################
+
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+, profiling ? false
+
+, runtimeShell
+, writeScriptBin
+, dockerTools
+
+, commonLib ? import ../lib.nix { inherit system crossSystem config profiling; }
+
+# The main contents of the image.
+, cardano-node
+
+# Get the current commit
+, gitrev ? commonLib.iohkNix.commitIdFromGitRepoOrZero ../.git
+
+# Other things to include in the image.
+, glibcLocales
+, iana-etc
+, cacert
+, bashInteractive
+, coreutils
+, utillinux
+, iproute
+, iputils
+, curl
+, socat
+
+# Used to generate the docker image names
+, repoName ? "inputoutput/cardano-node"
+}:
+
+let
+
+  startScript = writeScriptBin "start-cardano-node" ''
```

Comment:
I would prefer this to use scripts.<env>.node so parameters can be overridden with a customConfig parameter.

Reply:
We've removed the script section all together now.